scalardb-test/src/main/java/kelpie/scalardb/sensor/SensorChecker.java

```diff
@@ -0,0 +1,86 @@
+package kelpie.scalardb.sensor;
+
+import com.scalar.db.api.DistributedTransaction;
+import com.scalar.db.api.DistributedTransactionManager;
+import com.scalar.db.api.Result;
+import com.scalar.db.api.Scan;
+import com.scalar.db.exception.transaction.CrudException;
+import com.scalar.kelpie.config.Config;
+import com.scalar.kelpie.exception.PostProcessException;
+import com.scalar.kelpie.modules.PostProcessor;
+import io.github.resilience4j.retry.Retry;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import kelpie.scalardb.Common;
+
+public class SensorChecker extends PostProcessor {
+  private final DistributedTransactionManager manager;
+
+  public SensorChecker(Config config) {
+    super(config);
+    this.manager = SensorCommon.getTransactionManager(config);
+  }
+
+  @Override
+  public void execute() {
+    int startTimestamp = getIntFromPreviousState("start_timestamp");
+    int endTimestamp = getIntFromPreviousState("end_timestamp");
+
+    boolean isDuplicated = false;
+    for (int i = startTimestamp; i <= endTimestamp; i++) {
+      List<Result> results = readRecordsWithRetry(i);
+
+      boolean hasDuplicatedRevision = SensorCommon.hasDuplicatedRevision(results);
+      if (hasDuplicatedRevision) {
```

Comment:
Just a question. Is there a particular reason for not throwing an exception here ?

Reply:
I think it is better to check all timestamps for failure analysis rather than an exception that is thrown when the first duplicated revision is found.